endpoints/util.go

```diff
@@ -8,14 +8,19 @@ import (
 	"strings"
 )
 
```

Comment:
Potential optimization would be to track the last time the service was refreshed, only allow refreshing it so often. If it's brand new, refresh right away, but if we've already just checked for it for, maybe wait a bit before trying again.

Reply:
There are potentially complications with reserved shares there. I'd like to see how this goes and then further optimize if needed. But yes, we might be able to try and get more clever with reducing refreshes.